cookbook/profiler/data_collector.rst

```diff
@@ -16,70 +16,71 @@ Creating a custom data collector is as simple as implementing the
-         * @param Response   $response  A Response instance
-         * @param \Exception $exception An Exception instance
-         */
         function collect(Request $request, Response $response, \Exception $exception = null);
-
-        /**
-         * Returns the name of the collector.
-         *
-         * @return string The collector name
-         */
         function getName();
     }
 
-The value returned by ``getName()`` must be unique in the application. This value
-is also used to access the information later on (see :doc:`/cookbook/testing/profiling`
-for instance).
-
-The ``collect()`` method is responsible for storing the collected data in local
-properties.
+The
+:method:`Symfony\\Component\\HttpKernel\\DataCollector\\DataCollectorInterface::getName`
+method returns the name of the data collector and must be unique in the
+application. This value is also used to access the information later on (see
+:doc:`/cookbook/testing/profiling` for instance).
 
-.. caution::
-
-    As the profiler serializes data collector instances, you should not
-    store objects that cannot be serialized (like PDO objects), or you need
-    to provide your own ``serialize()`` method.
+The
+:method:`Symfony\\Component\\HttpKernel\\DataCollector\\DataCollectorInterface::collect`
+method is responsible for storing the collected data in local properties.
 
 Most of the time, it is convenient to extend
 :class:`Symfony\\Component\\HttpKernel\\DataCollector\\DataCollector` and
 populate the ``$this->data`` property (it takes care of serializing the
-``$this->data`` property)::
+``$this->data`` property). Imagine you create a new data collector that
+collects the method and accepted content types from the request::
+
+    // src/AppBundle/DataCollector/RequestCollector.php
+    namespace AppBundle\DataCollector;
 
-    // src/AppBundle/DataCollector/MyCollector.php
     use Symfony\Component\HttpKernel\DataCollector\DataCollector;
 
-    class MyCollector extends DataCollector
+    class RequestCollector extends DataCollector
     {
         public function collect(Request $request, Response $response, \Exception $exception = null)
         {
             $this->data = array(
-                'variable' => 'value',
+                'method' => $request->getMethod(),
+                'acceptable_content_types' => $request->getAcceptableContentTypes(),
             );
         }
 
-        public function getVariable()
+        public function getMethod()
         {
-            return $this->data['variable'];
+            return $this->data['method'];
+        }
+
+        public function getAcceptableContentTypes()
+        {
+            return $this->data['acceptable_content_types'];
         }
 
         public function getName()
         {
-            return 'app.my_collector';
+            return 'app.request_collector';
         }
     }
 
+The getters are added to give the template access to the collected information.
+
+.. caution::
```

Comment:
Moved the caution some lines below, as it was talking about overriding a `serialize()` method which part of the abstract data collector (introduced just above).